server/routes/charts.js

```diff
@@ -185,14 +185,26 @@ router.route('/charts/:chart_id')
           }
         }))
       const chartVariableColors = fieldValues.map(({ fieldLabelValueMap: { color }}) => color)
+      const groupTargetValuesByFieldValues = fieldValues
+        .reduce((currentGroup, nextGroup) => {
+          currentGroup[nextGroup.fieldLabelValueMap.label] = nextGroup.fieldLabelValueMap
+            .targetValues
+              .reduce((currentSites, nextSites) => {
+                currentSites[nextSites.site] = { value: nextSites.value }
+
+                return currentSites
+              }, {})
+          return currentGroup
+        }, {})
       const user = userFromRequest(req)
       const graph = { 
         chart_id, 
         data, 
         title: chartTitle, 
         description: chartDescription,
         legend,
-        chartVariableColors
+        chartVariableColors,
+        targetValuesMap: groupTargetValuesByFieldValues
       }
 
       return res.status(200).send(viewChartPage(user, graph))
```

Comment:
Scrolling up, there is a lot happening in this route file. I think it would be really valuable to break this out into smaller pieces/stages, something(s) that hit(s) the db & something(s) that work(s) with that data to manipulate/compose it into what we need. Then this file is really just responsible for routing, just takes the request, calls out to those functions and responds with the result.

  I think that would help with reading/reasoning about the code, and would start to move us towards having smaller chunks of logic that we could easily test in the future when we have capacity. It would also tie things together via interfaces which would make it easier to change pieces/stages of this chain going forward.

Have you thought about adding some sort of a Model to house these queries? Could be something as light as an object or could opt for something like Mongoose (I've used it in the past, it was okay) or maybe Prisma?(newer, I like it, I think works with Mongo?)

Reply:
[d28f2d6](https://github.com/AMP-SCZ/dpdash/pull/110/commits/d28f2d6b1be0f5c994babe2ddbf5581cbeed790c)
Thanks for the tip about splitting it up. 
@WMahoney09 got time today to chat a bit about Mongoose and Prisma?

Reply:
Np 🙂 yeah I have check-in and a planning meeting early afternoon, but otherwise free.